model/client4.go

```diff
@@ -3780,6 +3780,24 @@ func (c *Client4) GetPost(postId string, etag string) (*Post, *Response, error)
 	return &post, BuildResponse(r), nil
 }
 
+// GetPost gets a single post, including deleted.
+func (c *Client4) GetPostIncludeDeleted(postId string, etag string) (*Post, *Response, error) {
+	r, err := c.DoAPIGet(c.postRoute(postId)+"?include_deleted="+c.boolString(true), etag)
+	if err != nil {
+		return nil, BuildResponse(r), err
+	}
+	defer closeBody(r)
+
+	var post Post
+	if r.StatusCode == http.StatusNotModified {
+		return &post, BuildResponse(r), nil
+	}
+	if jsonErr := json.NewDecoder(r.Body).Decode(&post); jsonErr != nil {
+		return nil, nil, NewAppError("GetPostIncludeDeleted", "api.unmarshal_error", nil, jsonErr.Error(), http.StatusInternalServerError)
+	}
+	return &post, BuildResponse(r), nil
+}
+
```

Comment:
Note, decided to implement completely seperate client function for the specific case for including deleted as this was the pattern that I noticed throughout the file ([ex1](https://github.com/mattermost/mattermost-server/blob/54a88fb53268874aad436c77c72586ac24eb3e70/model/client4.go#L1919-L1950), [ex2](https://github.com/mattermost/mattermost-server/blob/54a88fb53268874aad436c77c72586ac24eb3e70/model/client4.go#L3313-L3343), etc...), as well as the fact that changing the current function would deprecate current users of the library.

That being said, not sure that this is the best pattern for scalability down the line, especially if more options are anticipated.

Reply:
It is fine as we don't want to have breaking changes :)